File: LanguAI.Backend/LanguAI.Backend/Core/Models/Language.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace LanguAI.Backend.Core.Models;

public class Language
{
    [Required]
    public int Id { get; set; }

    [Required]
    public string Code { get; set; }

    [Required]
    public string Name { get; set; }

    [Required]
    public string NameInHun {  get; set; }

    [JsonIgnore]
    public virtual ICollection<Learning> LearningLanguageOfLearnings { get; set; } = new List<Learning>();
    [JsonIgnore]
    public virtual ICollection<Learning> NativeLanguageOfLearnings { get; set; } = new List<Learning>();
    [JsonIgnore]
    public virtual ICollection<CardList> LearningLanguageOfCardLists { get; set; } = new List<CardList>();
    [JsonIgnore]
    public virtual ICollection<CardList> NativeLanguageOfCardLists { get; set; } = new List<CardList>();
}

File: LanguAI.Frontend/src/api/models/language.ts
/* tslint:disable */
/* eslint-disable */
import { CardList } from '../models/card-list';
import { Learning } from '../models/learning';
export interface Language {
  code: string;
  id: number;
  learningLanguageOfCardLists?: Array<CardList> | null;
  learningLanguageOfLearnings?: Array<Learning> | null;
  name: string;
  nameInHun: string;
  nativeLanguageOfCardLists?: Array<CardList> | null;
  nativeLanguageOfLearnings?: Array<Learning> | null;
}
